File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentTopicForm.ts
import { ServiceLookupEditor, LookupEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ContentTopicForm {
    ContentId: ServiceLookupEditor;
    ClassId: LookupEditor;
    SubjectId: LookupEditor;
    TopicId: LookupEditor;
    MediumId: LookupEditor;
    SortOrder: IntegerEditor;
}

export class ContentTopicForm extends PrefixedContext {
    static readonly formKey = 'Content.ContentTopic';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ContentTopicForm.init)  {
            ContentTopicForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = LookupEditor;
            var w2 = IntegerEditor;

            initFormType(ContentTopicForm, [
                'ContentId', w0,
                'ClassId', w1,
                'SubjectId', w1,
                'TopicId', w1,
                'MediumId', w1,
                'SortOrder', w2
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Content/ContentTopic/ContentTopicForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Content.Forms;

[FormScript("Content.ContentTopic")]
[BasedOnRow(typeof(ContentTopicRow), CheckNames = true)]
public class ContentTopicForm
{
    [HalfWidth]
    public int ContentId { get; set; }
    [HalfWidth]
    public int ClassId { get; set; }
    [HalfWidth]
    public int SubjectId { get; set; }
    [HalfWidth]
    public int TopicId { get; set; }
    [HalfWidth]
    public int MediumId { get; set; }
    [HalfWidth]
    public short SortOrder { get; set; }
}